### PR TITLE
[TENSOR MERGE] Use int64_t instead of ptrdiff_t for size / Rename flag to resizable_

### PR DESCRIPTION
**Pull request by cpuhrsch**

### CODE REVIEW

File: aten/src/ATen/Storage.h [hunk 1]
```diff
@@ -42,7 +42,7 @@ struct Storage {
   int getDevice() const {
     return storage->data_ptr.device().index();
   }
-  void clear_flag(char flag) {
+  void clear_flag(bool flag) {
```
Comment (gchanan) [inline aten/src/ATen/Storage.h]: having this be called flag when it means resizable doesn't make sense.

File: aten/src/TH/THStorageClass.cpp [hunk 2]
```diff
@@ -2,24 +2,24 @@
 
 THStorage::THStorage(
     at::ScalarType scalar_type,
-    ptrdiff_t size,
+    int64_t size,
     at::DataPtr data_ptr,
     at::Allocator* allocator,
-    char flag)
+    bool flag)
```
Comment (gchanan) [inline aten/src/TH/THStorageClass.cpp]: here too.

**Comment by ezyang** [comment]
```
22:31:01 /var/lib/jenkins/workspace/aten/src/ATen/test/atest.cpp:25
22:31:01 ...............................................................................
22:31:01 
22:31:01 /var/lib/jenkins/workspace/aten/src/ATen/test/atest.cpp:41: FAILED:
22:31:01   {Unknown expression after the reported line}
22:31:01 due to a fatal error condition:
22:31:01   SIGSEGV - Segmentation violation signal
```

Looks real.

File: aten/src/ATen/templates/StorageDerived.cpp [hunk 3]
```diff
@@ -10,19 +10,20 @@
...
 
-${Storage}::${Storage}(Context* context, size_t size, Allocator* allocator)
-  : Storage(nullptr), context(context) {
-  storage = ${THStorage}_newWithAllocator(${state,} size, allocator);
-  ${THStorage}_clearFlag(${state,} storage, TH_STORAGE_RESIZABLE);
+${Storage}::${Storage}(size_t size, Allocator* allocator)
+  : Storage(nullptr) {
+//  storage = ${THStorage}_newWithAllocator(${state,} size, allocator);
+  storage = new THStorage(ScalarType::${ScalarName}, size, allocator, TH_STORAGE_RESIZABLE);
+  THStorage_setResizable(storage, TH_STORAGE_RESIZABLE);
```
Comment (ezyang) [inline aten/src/ATen/templates/StorageDerived.cpp]: Don't use this constant anymore, right? Also, don't you want to say this is NOT resizable.

File: aten/src/TH/THGeneral.h.in [hunk 4]
```diff
@@ -5,6 +5,7 @@
 #define __STDC_FORMAT_MACROS
 #endif
 
+#include <stdbool.h>
```
Comment (ezyang) [inline aten/src/TH/THGeneral.h.in]: Hm?

File: aten/src/TH/generic/THStorage.cpp [hunk 5]
```diff
@@ -137,7 +132,7 @@ THStorage* THStorage_(newWithDataAndAllocator)(at::DataPtr&& data, ptrdiff_t siz
       size,
       std::move(data),
       allocator,
-      TH_STORAGE_REFCOUNTED | TH_STORAGE_RESIZABLE);
+      TH_STORAGE_RESIZABLE);
```
Comment (ezyang) [inline aten/src/TH/generic/THStorage.cpp]: You don't want to use this flag anymore either, right?

File: aten/src/TH/generic/THStorage.h [hunk 6]
```diff
@@ -20,8 +20,7 @@
    primitives??
  */
 
-#define TH_STORAGE_REFCOUNTED 1
-#define TH_STORAGE_RESIZABLE  2
+#define TH_STORAGE_RESIZABLE  1
```
Comment (ezyang) [inline aten/src/TH/generic/THStorage.h]: Just delete this constant altogether...

**Review by ezyang — CHANGES_REQUESTED** [review]
1. There are some mistakes in the refactoring; that's probably why the tests fail
2. Expunge `TH_STORAGE` entirely; just use a bool.

**Pull request closed without merging**